tests/unit/test_client.py

```diff
@@ -104,6 +107,23 @@ def to_bytes_iterator(content, chunk_size=64 * 1024, exception_to_raise=None):
             raise exception_to_raise
 
 
+def mock_response(expected_range, blob_length, content, exception=None, etag=blob_etag):
```

Comment:
For the signature, let's:

1. Update the name to `mock_download_response()` to make it a little more clear what type of response we are mocking
2. Remove the `blob_etag` default as that is a fixture and that cannot be used when this function is not a fixture. Instead we can just set the `etag` default to `None`.